software/firmware/europi.py

```diff
@@ -59,6 +59,18 @@
 CHAR_HEIGHT = 8
 
 
+
+# EuroPi error class
+
+class EuroPiError(Exception):
+    def __init__(self, error_code):
+        oled.fill(0)
+        oled.centre_text(f'Error {error_code}')

```

Comment:
[discussion] We can add ERROR_CODE as a class variable and allow sub-classed errors to provide consistent error codes.
```suggestion
    ERROR_CODE = 0  # EuroPiError subclasses define their own code.
    def __init__(self, message):
        super(message)
        oled.fill(0)
        oled.centre_text(f'Error {self.ERROR_CODE}')

```